tianshou/data/batch.py

```diff
@@ -500,6 +504,17 @@ def __getitem__(self, index: str | IndexType) -> Any:
             return new_batch
         raise IndexError("Cannot access item from empty Batch object.")
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, self.__class__):
+            return False
+
+        self.to_numpy()
```

Comment:
This modifies self and other inplace on a supposedly harmless equality check - this should never happen.

Starting a new thread about the questions you raised regarding `to_numpy`.

1. For other methods we have the convention that things ending in `_` are inplace. So it should really be that `to_numpy_` operates inplace and returns nothing, whereas `to_numpy()` doesn't change the original and returns a new batch. That's a breaking change, so you'd need to check all the places where `to_numpy` is currently being used.
2. I think the name `to_numpy` is ok for the moment, since all values of batch are either batches, tensors or arrays, right? Then it's clear that `to_numpy` would turn all tensors into arrays.

Reply:
Yes, thanks for bringing it up! We should generally reduce the number of inplace operations as much as possible